codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketDataFrameContainer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.http.websocketx;
+
+/**
+ * A container type that wraps some WebSocket data frame.
+ * This interface can be used for adding options to WebSocket data frame, that consumed by custom encoder.
+ */
+public interface WebSocketDataFrameContainer {
+
+    /**
+     * Returns the contained websocket data frame.
+     *
+     * @return the contained websocket data frame.
+     */
+    WebSocketFrame extractDataFrame();
```

Comment:
a bit awkward how this interface is only designed to contain data frames, but this method return type is `WebSocketFrame`. It looks like there is no `WebSocketDataFrame` interface though ... and I'm not sure if adding one really provides any value. @normanmaurer - WDYT?

Reply:
Seems a little bit unintuitive to me. How about adding a method like `WebSocketFrame.setExtensionFilter(WebSocketExtensionFilter filter)`, where `WebSocketExtensionFilter` is a predicate that allows a user skip the evaluation of a certain extension?